examples/intersection_observer/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+use seed::{prelude::*, *};
+use web_sys::{IntersectionObserver, IntersectionObserverEntry, IntersectionObserverInit};
+
+fn init(_: Url, orders: &mut impl Orders<Msg>) -> Model {
+    orders.after_next_render(|_| Msg::SetupObserver);
+    Model {
+        box_container: ElRef::new(),
+        red_box: ElRef::new(),
+        observer: None,
+        observer_callback: None,
+        observer_entries: None,
+    }
+}
+
+struct Model {
+    box_container: ElRef<web_sys::Element>,
+    red_box: ElRef<web_sys::Element>,
+    observer: Option<IntersectionObserver>,
+    observer_callback: Option<Closure<dyn Fn(Vec<JsValue>)>>,
+    observer_entries: Option<Vec<IntersectionObserverEntry>>,
+}
+
+enum Msg {
+    SetupObserver,
+    Observed(Vec<IntersectionObserverEntry>),
+}
+
+fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg>) {
+    match msg {
+        Msg::Observed(entries) => {
+            model.observer_entries = Some(entries);
+        }
+        Msg::SetupObserver => {
+            orders.skip();
+
+            // ---- observer callback ----
+            let sender = orders.msg_sender();
+            let callback = move |entries: Vec<JsValue>| {
+                let entries = entries
+                    .into_iter()
+                    .map(IntersectionObserverEntry::from)
+                    .collect();
+                sender(Some(Msg::Observed(entries)));
+            };
+            let callback = Closure::wrap(Box::new(callback) as Box<dyn Fn(Vec<JsValue>)>);
+
+            // ---- observer options ----
+            let mut options = IntersectionObserverInit::new();
+            options.threshold(&JsValue::from(1));
+            // ---- observer ----
+            let observer =
+                IntersectionObserver::new_with_options(callback.as_ref().unchecked_ref(), &options)
+                    .unwrap();
+
+            observer.observe(&model.red_box.get().unwrap());
```

Comment:
You don't see it here on GitHub, but it's badly formatted.
![image](https://user-images.githubusercontent.com/18517402/123395803-e5b10780-d5a0-11eb-8c4d-8476f87295a2.png)
Please run `cargo fmt`. Another reason why we need `Makefile.toml` - we want to automatically lint the example by CI.